app/models/concerns/host_common.rb

```diff
@@ -92,15 +92,28 @@ def puppetca_exists?
     !!(puppet_ca_proxy && puppet_ca_proxy.url.present?)
   end
 
-  # no need to store anything in the db if the entry is plain "puppet"
-  # If the system is using smart proxies and the user has run the smartproxy:migrate task
-  # then the puppetmaster functions handle smart proxy objects
+  def puppetmaster_uri
+    return nil unless puppet_proxy
+    url = puppet_proxy.setting('Puppet', 'puppet_url')
+    url ||= "https://#{puppet_proxy}:8140"
+    URI(url)
+  end
+
+  # Deprecated way to get the puppetmaster
   def puppetmaster
-    puppet_proxy.to_s
```

Comment:
ruby
```
Foreman::Deprecation.deprecation_warning('1.24', 'explain what and why is deprecated and what should be used instead.')
```

Reply:
If you add a deprecation in what's used as a macro, will that continue to work?

Reply:
I think it should, deprecation should be only print out to log, or did have specific concern in mind?

Reply:
Yes, except tests will start to fail, therefore it must be removed from all tests.